common/src/main/java/org/apache/apex/common/util/JarHelper.java

```diff
@@ -107,6 +114,73 @@ public String getJar(Class<?> jarClass)
     return jar;
   }
 
+  public String getJar(Class<?> jarClass)
+  {
+    return getJar(jarClass, true);
+  }
+
+  /**
+   * Returns a full path to the jar-file that contains the given class and all full paths to dependent jar-files
+   * that are defined in the property "apex-dependencies" of the manifest of the root jar-file.
+   * If the class is an independent file the method makes jar file from the folder that contains the class
+   * @param jarClass Class
+   * @param makeJarFromFolder True if the method should make jar from folder that contains the independent class
+   * @param addJarDependencies True if the method should include dependent jar files
+   * @return List of names of the jar-files
+   */
+  public List<String> getJars(Class<?> jarClass, boolean makeJarFromFolder, boolean addJarDependencies)
```

Comment:
How is uniqueness guaranteed? You can't rely on apex-dependencies to be unique. There is no need to implement Set or List, the complexity of implementation is hidden.